### PR TITLE
SDK changes to support etag for VM/VMSS/VMSSVM

### PR DESCRIPTION
**Pull request by shivisaxena21**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachineScaleSetVMsOperations.cs [hunk 1]
```diff
@@ -140,7 +149,7 @@ public partial interface IVirtualMachineScaleSetVMsOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse<VirtualMachineScaleSetVM>> UpdateWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, VirtualMachineScaleSetVM parameters, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse<VirtualMachineScaleSetVM>> UpdateWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, VirtualMachineScaleSetVM parameters, string ifMatch = default(string), string ifNoneMatch = default(string), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (archerzz) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachineScaleSetVMsOperations.cs:152]: Inserting new parameters is a breaking change. Need to write customization codes to keep backward compatibility. See https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/compute/Microsoft.Azure.Management.Compute/src/Customizations for example.

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GrantAccessData.cs [hunk 2]
```diff
@@ -36,15 +36,11 @@ public GrantAccessData()
         /// SAS access expires.</param>
         /// <param name="getSecureVMGuestStateSAS">Set this flag to true to get
         /// additional SAS for VM guest state</param>
-        /// <param name="fileFormat">Used to specify the file format when
```
Comment (archerzz) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/GrantAccessData.cs:39]: Removing `fileFormat` is a breaking change. You can add back the constructor and `FileFormat` property, but make it useless in reality.

**Review by archerzz — CHANGES_REQUESTED** [review]
There are two types of breaking changes which need to handle:
- Insertion of `string ifMatch = default(string), string ifNoneMatch = default(string)`
- Deletion of `FileFormat`.

**Comment by grizzlytheodore** [comment]
closing this PR since this is Track 1 and it is deprecated

**Pull request closed without merging**